packages/js/src/plugins/auctionHouseModule/executeSale.ts

```diff
@@ -0,0 +1,155 @@
+  response: SendAndConfirmTransactionResponse;
+};
+
+// -----------------
+// Handler
+// -----------------
+
+export const executeSaleOperationHandler: OperationHandler<ExecuteSaleOperation> =
+  {
+    handle: async (operation: ExecuteSaleOperation, metaplex: Metaplex) =>
+      executeSaleBuilder(metaplex, operation.input).sendAndConfirm(
+        metaplex,
+        operation.input.confirmOptions
+      ),
+  };
+
+// -----------------
+// Builder
+// -----------------
+
+export type ExecuteSaleBuilderParams = Omit<
+  ExecuteSaleInput,
+  'confirmOptions'
+> & {
+  instructionKey?: string;
+};
+
+export type ExecuteSaleBuilderContext = Omit<ExecuteSaleOutput, 'response'>;
+
+export const executeSaleBuilder = (
+  metaplex: Metaplex,
+  params: ExecuteSaleBuilderParams
+): TransactionBuilder<ExecuteSaleBuilderContext> => {
+  // Data.
+  const auctionHouse = params.auctionHouse;
+  const tokens = params.tokens ?? token(1);
+  const priceBasisPoint = params.price?.basisPoints ?? 0;
+  const price = auctionHouse.isNative
+    ? lamports(priceBasisPoint)
+    : amount(priceBasisPoint, auctionHouse.treasuryMint.currency);
+
+  // Accounts.
+  const seller = params.seller ?? (metaplex.identity() as Signer);
+  const tokenAccount = findAssociatedTokenAccountPda(
+    params.mintAccount,
+    toPublicKey(seller)
+  );
+  const buyerTokenAccount = findAssociatedTokenAccountPda(
+    params.mintAccount,
+    toPublicKey(params.buyer)
+  );
+  const metadata = findMetadataPda(params.mintAccount);
+  const escrowPayment = findAuctionHouseBuyerEscrowPda(
+    auctionHouse.address,
+    toPublicKey(params.buyer)
+  );
+  const freeTradeState = findAuctionHouseTradeStatePda(
+    auctionHouse.address,
+    toPublicKey(seller),
+    auctionHouse.treasuryMint.address,
+    params.mintAccount,
+    lamports(0).basisPoints,
+    tokens.basisPoints,
+    tokenAccount
+  );
+  const programAsSigner = findAuctionHouseProgramAsSignerPda();
+
+  const accounts = {
+    buyer: toPublicKey(params.buyer),
+    seller: toPublicKey(seller),
+    tokenAccount,
+    tokenMint: params.mintAccount,
+    metadata,
+    treasuryMint: auctionHouse.treasuryMint.address,
+    escrowPaymentAccount: escrowPayment,
+    sellerPaymentReceiptAccount: toPublicKey(seller),
+    buyerReceiptTokenAccount: buyerTokenAccount,
+    authority: auctionHouse.authorityAddress,
+    auctionHouse: auctionHouse.address,
+    auctionHouseFeeAccount: auctionHouse.feeAccountAddress,
+    auctionHouseTreasury: auctionHouse.treasuryAccountAddress,
+    buyerTradeState: params.buyerTradeState,
+    sellerTradeState: params.sellerTradeState,
+    freeTradeState,
+    programAsSigner,
+  };
+
+  // Args.
+  const args = {
+    freeTradeStateBump: freeTradeState.bump,
+    escrowPaymentBump: escrowPayment.bump,
+    programAsSignerBump: programAsSigner.bump,
+    buyerPrice: price.basisPoints,
+    tokenSize: tokens.basisPoints,
+  };
+
+  return TransactionBuilder.make<ExecuteSaleBuilderContext>().add({
```

Comment:
I believe we should be sending another instruction after executing the sale to create a `Purchase` receipt. This means we need to implement the `Purchase` model accordingly.